### ISSUE TITLE
How do I install this in a venv? 

### ISSUE THREAD

**Issue by NemukaPrime** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#2303]
Help needed! Been searching online for hours. I have other WebUIs and they seem to have their own Python venvs.
When I try to install RVC-Project, it fails at `pip install -r requirements.txt`.

Although I'm not sure if I did the previous step correctly (installing PyTorch) since I'm not sure how to specify the CUDA version. I think other WebUIs I have installed have PyTorch but I'm not sure if it's system wide. I also don't want to change anything that might break the other WebUIs I have. Also, if I go to the linked https://download.pytorch.org/whl/cu117, it just has a list of what look like packages or something. I'm not sure how to specify the cuda version.

Using the `curl -sSL https://install.python-poetry.org | python3 -` line also gave an error, so after looking through the Issues, I saw that pip can be used to install them. 

Using pip, there was an error afterward that said:

> "WARNING: Ignoring invalid distribution ~ip (C:\Python312\Lib\site-packages)
> Collecting joblib>=1.1.0 (from -r requirements.txt (line 1))
>   Using cached joblib-1.4.2-py3-none-any.whl.metadata (5.4 kB)
> Collecting numba==0.56.4 (from -r requirements.txt (line 2))
>   Using cached numba-0.56.4.tar.gz (2.4 MB)
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... error
>   error: subprocess-exited-with-error
> 
>   × Getting requirements to build wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [23 lines of output]
>       C:\Users\User\AppData\Local\Temp\pip-install-5cvd45uj\numba_714aed0fecfb4496a7a06531d1c04fcf\versioneer.py:335: SyntaxWarning: invalid escape sequence '\s'
>         LONG_VERSION_PY['git'] = '''
>       Traceback (most recent call last):
>         File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
>           main()
>         File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
>           json_out['return_val'] = hook(**hook_input['kwargs'])
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
>           return hook(config_settings)
>                  ^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\User\AppData\Local\Temp\pip-build-env-x7dgk80c\overlay\Lib\site-packages\setuptools\build_meta.py", line 332, in get_requires_for_build_wheel
>           return self._get_build_requires(config_settings, requirements=[])
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\Usere\AppData\Local\Temp\pip-build-env-x7dgk80c\overlay\Lib\site-packages\setuptools\build_meta.py", line 302, in _get_build_requires
>           self.run_setup()
>         File "C:\Users\User\AppData\Local\Temp\pip-build-env-x7dgk80c\overlay\Lib\site-packages\setuptools\build_meta.py", line 503, in run_setup
>           super().run_setup(setup_script=setup_script)
>         File "C:\Users\User\AppData\Local\Temp\pip-build-env-x7dgk80c\overlay\Lib\site-packages\setuptools\build_meta.py", line 318, in run_setup
>           exec(code, locals())
>         File "<string>", line 51, in <module>
>         File "<string>", line 48, in _guard_py_ver
>       RuntimeError: Cannot install on Python version 3.12.4; only versions >=3.7,<3.11 are supported.
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: subprocess-exited-with-error
> 
> × Getting requirements to build wheel did not run successfully.
> │ exit code: 1
> ╰─> See above for output.
> 
> note: This error originates from a subprocess, and is likely not a problem with pip."
> 

If I'm not mistaken, this means that I need another Python version, right? I'm not well-versed in these things, but from what I know, installing these in a venv would help. 

It's confusing since other WebUIs seem to do this automatically, but it would be helpful if there were instructions for installing the RVC-Project in a venv as well. If I'm totally wrong in my assumptions, please let me know as I'm new to this. Thanks for reading.

**Comment by AznamirWoW** [comment]
you are not gonna have good experience with python 3.12. Better use 3.10.15.

**Comment by abhis1n** [comment]
Python 3.12 is not supported, I'm using 3.10 and it works fine. 

Make a virtual environment using Python 3.10 and also ensure pip version is below 24.1. Then you can proceed to download [pytorch](https://pytorch.org/get-started/locally/) and then finally use `pip install -r requirements.txt`.

**Comment by NemukaPrime** [comment]
> Python 3.12 is not supported, I'm using 3.10 and it works fine.
> 
> Make a virtual environment using Python 3.10 and also ensure pip version is below 24.1. Then you can proceed to download [pytorch](https://pytorch.org/get-started/locally/) and then finally use `pip install -r requirements.txt`.

Thank you for the replies. I'm on Windows 10. Would you know how I can make a virtual environment using Python 3.10? (Or where I can learn how to do so?)

Thank you.

**Comment by abhis1n** [comment]
1. Install Python 3.10
2. Open cmd in it's directory and run this command- `python -m venv [path-to-RVC]\runtime`

This will make a directory named 'runtime' in the RVC folder.
The code looks like this on my machine-

> C:\Python310>python -m venv D:\Retrieval-based-Voice-Conversion-WebUI\runtime

3. Open powershell in RVC folder
4. Run this command to activate virtual environment- `.\runtime\Scripts\Activate.ps1`

from here, you can follow the previous steps.

**Comment by NemukaPrime** [comment]
Hello, thank you for the help. After a few more hours of trying and troubleshooting, I can see the webui now thanks to your instructions, but now it's saying something about documentation.py.

Here's what I was able to do (I'm putting the entirety of it just in case but tl;dr below):
**tl;dr There's something wrong in Step 6 when I try to run go-web.bat**

1. Already had Python 3.10.6 installed, so I didn't reinstall it. I moved the PATH user variables and system variables up for the Python310 folder since I had miniconda3 with Python 3.12.
2. Running one of the commands was showing a folder that doesn't exist "E:\Codes\<something>", so I renamed the preexisting runtime folder to "runtime old".
3. I opened cmd admin and checked the Python version (3.10.6) and pip version (22.2.1).
4. I used the command you gave, `python -m venv [path-to-RVC]\runtime` and then navigated to runtime\scripts. I activated using command prompt "activate". 
5. I installed pytorch with the appropriate CUDA version found through the nvcc command.
After that, I did the `pip install -r requirements.txt` command, then downloaded the models using "python tools/download_models.py"
6. I changed the go-web.bat to use Python.exe in runtime\Scripts because my runtime didn't have python inside it. Python.exe was in the runtime\Scripts folder.
I suspect something went wrong in Step 6 because Python.exe wasn't in the runtime folder but in Scripts. 

After this, I was able to open the webui using go-web.bat, but it doesn't work after selecting an audio file and inferencing voice. cmd says this:

> E:\AI-Voice\RVC1006Nvidia\runtime\lib\site-packages\gradio_client\documentation.py:106: UserWarning: Could not get documentation group for <class 'gradio.mix.Parallel'>: No known documentation group for module 'gradio.mix'
>   warnings.warn(f"Could not get documentation group for {cls}: {exc}")
> E:\AI-Voice\RVC1006Nvidia\runtime\lib\site-packages\gradio_client\documentation.py:106: UserWarning: Could not get documentation group for <class 'gradio.mix.Series'>: No known documentation group for module 'gradio.mix'
>   warnings.warn(f"Could not get documentation group for {cls}: {exc}")
> 2024-09-16 19:54:49 | INFO | __main__ | Use Language: en_US

Help would be appreciated, preferably without having to redownload the model files. Thank you.

**Comment by abhis1n** [comment]
It is just a warning, you can still use RVC without any issues. Also, if the inference fails for large audio files [this](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/1054#issuecomment-2341864342) might help you.

**Comment by NemukaPrime** [comment]
Thank you, it's working now!

I had a problem mentioned in this issue #2048, and downgrading av in the venv as AleksisVorobjovs said through `pip install --force-reinstall -v "av==11.0.0"` made it finally work. Thank you again for the instructions!
